examples/with-styled-components/pages/_document.js

```diff
@@ -0,0 +1,29 @@
+import Document, { Head, Main, NextScript } from 'next/document'
+import styleSheet from 'styled-components/lib/models/StyleSheet'
+
+export default class MyDocument extends Document {
+  static async getInitialProps (ctx) {
+    const renderPage = () => {
+      return ctx.renderPage()
+    }
+
+    const props = await Document.getInitialProps({ ...ctx, renderPage })
+    const style = styleSheet.rules().map(rule => rule.cssText).join('\n')
+    return { ...props, style }
+  }
```

Comment:
@rauchg I noticed all examples on Wiki are calling `Document.getInitialProps`. What do you think about this API ?